spec/machine/AARCH64/MachineOps.thy

```diff
@@ -194,21 +184,25 @@ definition setFPUState :: "fpu_state \<Rightarrow> unit user_monad" where
 \<comment> \<open>FIXME FPU: The underlying register is saved and restored when switching between VCPUs, so this
      will probably need to be moved out of machine\_rest\_monad.\<close>
 consts' enableFpuEL01_impl :: "unit machine_rest_monad"
+consts' enableFpuEL01_val :: "machine_word \<Rightarrow> machine_word"
 definition enableFpuEL01 :: "unit machine_monad" where
-  "enableFpuEL01 \<equiv> machine_op_lift enableFpuEL01_impl"
+  "enableFpuEL01 \<equiv> do
+     modify (vcpu_state_update (vcpu_regs_update (\<lambda>r. r(VCPURegCPACR := enableFpuEL01_val (r VCPURegCPACR)))));
+     machine_op_lift enableFpuEL01_impl
+   od"
```

Comment:
Can you explain what's going on here? It's not clear to me how the _val is being used, and whether this is enough to remove the FIXME.

Reply:
`enableFpuEL01_val` abstracts from how the current value is modified. FIXME can be removed, thanks for spotting